### ISSUE TITLE
Patch for Python3 compatibility

### ISSUE THREAD

**Issue by CDLuminate** [fartashf/vsepp#12]
```patch
diff --git a/data.py b/data.py
index 913ea16..520c38d 100644
--- a/data.py
+++ b/data.py
@@ -221,13 +221,12 @@ class PrecompDataset(data.Dataset):
     def __getitem__(self, index):
         # handle the image redundancy
         img_id = index/self.im_div
-        image = torch.Tensor(self.images[img_id])
+        image = torch.Tensor(self.images[int(img_id)])
         caption = self.captions[index]
         vocab = self.vocab
 
         # Convert caption (string) to word ids.
-        tokens = nltk.tokenize.word_tokenize(
-            str(caption).lower().decode('utf-8'))
+        tokens = nltk.tokenize.word_tokenize(str(caption).lower())
         caption = []
         caption.append(vocab('<start>'))
         caption.extend([vocab(token) for token in tokens])
diff --git a/evaluation.py b/evaluation.py
index 7e5da4e..9171f85 100644
--- a/evaluation.py
+++ b/evaluation.py
@@ -57,7 +57,7 @@ class LogCollector(object):
         """Concatenate the meters in one log line
         """
         s = ''
-        for i, (k, v) in enumerate(self.meters.iteritems()):
+        for i, (k, v) in enumerate(self.meters.items()):
             if i > 0:
                 s += '  '
             s += k + ' ' + str(v)
@@ -66,7 +66,7 @@ class LogCollector(object):
     def tb_log(self, tb_logger, prefix='', step=None):
         """Log using tensorboard
         """
-        for k, v in self.meters.iteritems():
+        for k, v in self.meters.items():
             tb_logger.log_value(prefix + k, v.val, step=step)
 
 
```

**Comment by CDLuminate** [comment]
Fixes https://github.com/fartashf/vsepp/issues/5
The patch is not proposed as a PR because there are python2 users. Putting it here might be helpful to py3 users.

**Comment by CDLuminate** [comment]
The original patch doesn't solve the failure on evaluation. This is the updated patch
```patch
diff --git a/data.py b/data.py
index 913ea16..520c38d 100644
--- a/data.py
+++ b/data.py
@@ -221,13 +221,12 @@ class PrecompDataset(data.Dataset):
     def __getitem__(self, index):
         # handle the image redundancy
         img_id = index/self.im_div
-        image = torch.Tensor(self.images[img_id])
+        image = torch.Tensor(self.images[int(img_id)])
         caption = self.captions[index]
         vocab = self.vocab
 
         # Convert caption (string) to word ids.
-        tokens = nltk.tokenize.word_tokenize(
-            str(caption).lower().decode('utf-8'))
+        tokens = nltk.tokenize.word_tokenize(str(caption).lower())
         caption = []
         caption.append(vocab('<start>'))
         caption.extend([vocab(token) for token in tokens])
diff --git a/evaluation.py b/evaluation.py
index 7e5da4e..54547f8 100644
--- a/evaluation.py
+++ b/evaluation.py
@@ -57,7 +57,7 @@ class LogCollector(object):
         """Concatenate the meters in one log line
         """
         s = ''
-        for i, (k, v) in enumerate(self.meters.iteritems()):
+        for i, (k, v) in enumerate(self.meters.items()):
             if i > 0:
                 s += '  '
             s += k + ' ' + str(v)
@@ -66,7 +66,7 @@ class LogCollector(object):
     def tb_log(self, tb_logger, prefix='', step=None):
         """Log using tensorboard
         """
-        for k, v in self.meters.iteritems():
+        for k, v in self.meters.items():
             tb_logger.log_value(prefix + k, v.val, step=step)
 
 
@@ -209,7 +209,7 @@ def i2t(images, captions, npts=None, measure='cosine', return_ranks=False):
     Captions: (5N, K) matrix of captions
     """
     if npts is None:
-        npts = images.shape[0] / 5
+        npts = int(images.shape[0] / 5)
     index_list = []
 
     ranks = numpy.zeros(npts)
@@ -262,7 +262,7 @@ def t2i(images, captions, npts=None, measure='cosine', return_ranks=False):
     Captions: (5N, K) matrix of captions
     """
     if npts is None:
-        npts = images.shape[0] / 5
+        npts = int(images.shape[0] / 5)
     ims = numpy.array([images[i] for i in range(0, len(images), 5)])
 
     ranks = numpy.zeros(5 * npts)
```

**Comment by fartashf** [comment]
Closed by #13. Thanks.
